backend/src/test/scala/cromwell/backend/BackendActorSpec.scala

```diff
@@ -0,0 +1,156 @@
+            }
+          case unknown => fail(s"Response is not of type ArrayBuffer. Response: $unknown")
+        }
+      }
+    }
+
+    "execute afterAll just after backend actor instance is killed." in {
+      within(Timeout) {
+        testWorkflowActor ! StopBackendActor
+        Thread.sleep(50) //Giving time to WorkflowActor to receive the event.
+        testWorkflowActor ! GetEvents
+        expectMsgPF() {
+          case backendActorEvents: EventList =>
+            if (!backendActorEvents.events.contains(ValidationSucceeded)
+              || !backendActorEvents.events.contains(BeforeAllSucceeded)
+              || !backendActorEvents.events.contains(AfterAllSucceeded)) {
+              fail(s"All expected events were not raised. Expected events: ValidationSucceed, BeforeAllSucceeded and AfterAllSucceeded.")
+            }
+          case unknown => fail(s"Response is not of type ArrayBuffer. Response: $unknown")
+        }
+      }
+    }
+  }
+}
+
+object TestWorkflowActor {
+
+  trait TestWorkflowActorMessage
+
+  case object GetEvents extends TestWorkflowActorMessage
+
+  case object StopBackendActor extends TestWorkflowActorMessage
+
+  case class EventList(events: List[BackendActorEvent])
+
+  def props(): Props = Props(new TestWorkflowActor(
+    WorkflowDescriptor(UUID.fromString("8d772774-f76c-11e5-9ce9-5e5517507c66"))))
+
+}
+
+class TestWorkflowActor(workflowDescriptor: WorkflowDescriptor) extends Actor {
+  val testBackendActor = context.actorOf(TestBackendActor.props(workflowDescriptor))
+  var listOfEvents = ArrayBuffer[BackendActorEvent]()
+
+  override def receive = {
+    case GetEvents =>
+      sender() ! EventList(listOfEvents.toList)
+    case event: BackendActorEvent =>
+      listOfEvents += event
+    case StopBackendActor =>
+      testBackendActor ! PoisonPill
+  }
+}
+
+object TestBackendActor {
+
+  def props(workflowDescriptor: WorkflowDescriptor): Props = Props(new TestBackendActor(
+    workflowDescriptor))
+
+}
+
+class TestBackendActor(val workflowDescriptor: WorkflowDescriptor) extends BackendActor {
+  /**
+    * Tries to recover last status and information on an on going job in the backend.
+    *
+    * @return An RecoverEvent with the result.
+    */
+  override def recover(): RecoverEvent = ???
+
+  /**
+    * Executes a job base on a job description.
+    *
+    * @param jobDescriptor All information needed to execute a task in the backend.
+    * @return An ExecutionEvent with the result.
+    */
+  override def execute(jobDescriptor: JobDescriptor): Future[ExecutionEvent] = ???
+
+  /**
+    * Stops all job execution.
+    *
+    * @return An AbortAllEvent with the result.
+    */
+  override def abortAll(): AbortAllEvent = ???
+
+  /**
+    * Executes validation on workflow descriptor in order to see if the workflow can be executed by the backend.
+    */
+  override def validate(): Unit = {
+    context.parent ! ValidationSucceeded
```

Comment:
I just created this test as an example of how it would looks like preStart and postStop in BackendActor.